PixelHunter1995/Polygon.cs

```diff
@@ -242,5 +242,69 @@ public void Draw(GraphicsDeviceManager graphics)
                     drawIndices.Length - 1);
             }
         }
+
+        public bool HasCommonEdgeWith(Polygon other)
```

Comment:
Feels like this could be improved by not checking every other.vertices ('m' for brevity) for every vertices ('n') (nested loops), but instead iterate both n and m (not nested) to populate a map/histogram (though the fact that they are float coords might make this vastly harder) with the 2 adjecents/edges (`<Coord->List<Pair<Coord>>>` or ideally `<Edge -> int>`), which is then checked for vertices at same Coord (list > 1), whose adjecents/edges are then checked (make sure to skip/remove edges already checked) if they are the same.
Ideally the histogram would not be on vertices (Coord), but rather on edges (Coord pair), as then nothing has to be checked after a key has been found to already exist in map (allows for easy early termination too). Again, not all that easy to do, as it not only involves floating coords, but also should be unordered ({A, B} == {B, A}).

Should result in n+m instead of n*m.

Reply:
Ironically, one of the easier ways to achieve unordered pairs/edges, is to actually order (sort) the vertices when set. Though it is also possible to make the equals check both directions, and come up with a hash that doesn't differ based on order. I have an old project with code for the latter, though I would not confidently state that the hash-method is good (`vertice1.hash + vertice2.hash`...)

Reply:
Yeah, that might be a nice way to do it. But I think my way here is clear enough, and time is not really of an essence here.

However, `CombineWith()` is some messy shit... So that would be nice to rewrite, if we come up with something good.